google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineRuntimeInstance.java

```diff
@@ -61,32 +63,42 @@ public void deploy(@NotNull final DeploymentTask<AppEngineDeploymentConfiguratio
       return;
     }
 
-    AppEngineHelper appEngineHelper = new CloudSdkAppEngineHelper(task.getProject());
-
     AppEngineDeploymentConfiguration deploymentConfig = task.getConfiguration();
 
-    final CancellableRunnable deployRunner =  appEngineHelper.createDeployRunner(
-        logManager.getMainLoggingHandler(),
-        task.getSource(),
-        deploymentConfig,
-        callback);
+    AppEngineEnvironment environment =
+        AppEngineEnvironment.valueOf(deploymentConfig.getEnvironment());
+    if (environment == AppEngineEnvironment.APP_ENGINE_FLEX
+        && Strings.isNullOrEmpty(deploymentConfig.getYamlPath())) {
+      callback.errorOccurred(GctBundle.getString("appengine.deployment.error.staging.yaml")
+          + "\n" + GctBundle.getString("appengine.deployment.error.suggestflexfacet"));
```

Comment:
is it possible to make it more explicit / readable by guarding this message under a check for a missing deployable? I believe that we have a check like that somewhere, although I'm not sure if its where you need it for this.